test: improve lib/internal/readline/promises.js coverage

fossamagna opened the pull request:
<!--
Before submitting a pull request, please read
https://github.com/nodejs/node/blob/HEAD/CONTRIBUTING.md.

Commit message formatting guidelines:
https://github.com/nodejs/node/blob/HEAD/doc/contributing/pull-requests.md#commit-message-guidelines

For code changes:
1. Include tests for any bug fixes or new features.
2. Update documentation if relevant.
3. Ensure that `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
-->

This improves a test coverage in `lib/internal/readline/promises.js`

ref: https://coverage.nodejs.org/coverage-419f02ba1f00cac3/lib/internal/readline/promises.js.html

This validates that `autoCommit` option work correctly.

aduh95 commented on test/parallel/test-readline-promises-csi.mjs:
```diff
@@ -161,3 +161,85 @@ class TestWritable extends Writable {
+    const test = async (dir, data) => {
+      writable.data = '';
+      await readline.clearLine(dir);
+      return new Promise((resolve) => {
+        process.nextTick(() => {
+          assert.deepStrictEqual(writable.data, data);
+          resolve();
+        });
```
```suggestion
    process.nextTick(() => assert.deepStrictEqual(writable.data, data));
```

fossamagna commented:
@aduh95 
Thank you for your review.

I removed test function in response to take your suggestion, but test got following error.

```
AssertionError [ERR_ASSERTION]: Expected values to be strictly deep-equal:
+ actual - expected

+ '\x1B[1K\x1B[0K'
- '\x1B[0K'
        ^
```

When use autoCommit option, readline write to stream in callback function of process.nextTick. If we removed test function, before readline write to stream invocation of `clearLine` function in `for-of` loop will be finished.

fossamagna commented:
@aduh95 Thank you very much for your suggestion.
It works like a charm and have been more readable.

aduh95 commented on test/parallel/test-readline-promises-csi.mjs:
```diff
@@ -161,3 +162,61 @@ class TestWritable extends Writable {
+  const readline = new Readline(writable, { autoCommit: true });
+
+  await readline.clearScreenDown();
+  process.nextTick(() => assert.deepStrictEqual(writable.data, CSI.kClearScreenDown));
```
```suggestion
  await setImmediate(); // wait for next tick as auto commit is asynchronous.
  assert.deepStrictEqual(writable.data, CSI.kClearScreenDown);
```

aduh95 commented on test/parallel/test-readline-promises-csi.mjs:
```diff
@@ -161,3 +162,61 @@ class TestWritable extends Writable {
+    ]) {
+    writable.data = '';
+    readline.clearLine(dir);
+    assert.deepStrictEqual((await setImmediate(writable)).data, data);
```
```suggestion
    await setImmediate(); // wait for next tick as auto commit is asynchronous.
    assert.deepStrictEqual(writable.data, data);
```

aduh95 commented on test/parallel/test-readline-promises-csi.mjs:
```diff
@@ -161,3 +162,61 @@ class TestWritable extends Writable {
+    ]) {
+    writable.data = '';
+    readline.moveCursor(x, y);
+    assert.deepStrictEqual((await setImmediate(writable)).data, data);
```
```suggestion
    await setImmediate(); // wait for next tick as auto commit is asynchronous.
    assert.deepStrictEqual(writable.data, data);
```

aduh95 commented on test/parallel/test-readline-promises-csi.mjs:
```diff
@@ -161,3 +162,61 @@ class TestWritable extends Writable {
+    ]) {
+    writable.data = '';
+    readline.cursorTo(x, y);
+    assert.deepStrictEqual((await setImmediate(writable)).data, data);
```
```suggestion
    await setImmediate(); // wait for next tick as auto commit is asynchronous.
    assert.deepStrictEqual(writable.data, data);
```

juanarbol commented:
Dependent of https://github.com/nodejs/node/pull/37947
